skywalking/plugins/__init__.py

```diff
@@ -38,7 +41,65 @@ def install():
             continue
         logger.debug('installing plugin %s', modname)
         plugin = importer.find_module(modname).load_module(modname)
+
+        logger.debug('checking version for plugin %s', modname)
+        supported = pkg_version_check(plugin, modname)
+        if not supported:
+            continue
+
         if not hasattr(plugin, 'install') or inspect.ismethod(getattr(plugin, 'install')):
             logger.warning('no `install` method in plugin %s, thus the plugin won\'t be installed', modname)
             continue
         plugin.install()
+
+
+_operators = {
+    '<': lambda cv, ev: cv < ev,
+    '<=': lambda cv, ev: cv < ev or cv == ev,
+    '=': lambda cv, ev: cv == ev,
+    '>=': lambda cv, ev: cv > ev or cv == ev,
+    '>': lambda cv, ev: cv > ev,
+    '!=': lambda cv, ev: cv != ev
+}
+
+
+def pkg_version_check(plugin, modname):
+    supported = True
+
+    # no version rules was set, no checks
+    if not hasattr(plugin, "version_rule"):
+        return supported
+
+    pkg_name = plugin.version_rule.get("name")
+    rules = plugin.version_rule.get("rules")
+
+    try:
+        current_pkg_version = pkg_resources.get_distribution(pkg_name).version
+    except pkg_resources.DistributionNotFound:
+        supported = False
+        logger.warning("plugin %s didn\'t find the corresponding package %s, thus won't be installed",
+                       modname, pkg_name)
+        return supported
+
+    # check all rules
+    for rule in rules:
+        idx = 2 if rule[1] == '=' else 1
+        symbol = rule[0:idx]
+        expect_pkg_version = rule[idx:]
+
+        current_version = version.parse(current_pkg_version)
+        expect_version = version.parse(expect_pkg_version)
+        f = _operators.get(symbol) or None
+
+        # version rule parse error, take it as no more version rules and return True.
+        if not f:
+            logger.warning("plugin %s version rule %s error. only allow >,>=,=,<=,<,!= symbols", modname, rule)
+            return supported
+
+        if not f(current_version, expect_version):
+            supported = False
+            logger.warning("plugin %s need package %s version follow rules %s ,current version " +
+                           "is %s, thus won\'t be installed", modname, pkg_name, str(rules), current_pkg_version)
```

Comment:
Move this kind of message to where the invoker calls this method, I mark [1] in the previous comment

<img width="914" alt="image" src="https://user-images.githubusercontent.com/15965696/89915973-97fe0b80-dc29-11ea-80cf-5179c7482a02.png">